fix: fix plusAssign ambiguity (#2481)
JetBrains/kotlin-web-site: docs/topics/operator-overloading.md

For the assignment operations, for example `a += b`, the compiler performs the following steps:

* If the function from the right column is available:

>   * If the corresponding binary function (that means `plus()` for `plusAssign()`) is available too, `a` is a mutable variable, and the return type of `plus` is the same as the type of `a`, report error (ambiguity).

Reviewer: ```suggestion
  * If the corresponding binary function (that means `plus()` for `plusAssign()`) is available too, `a` is a mutable variable, and the return type of `plus` is a subtype of the type of `a`, report an error (ambiguity).
```
or "... the return type of `plus` is assignable to the type of `a` ..." which I suppose is equivalent.
